src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -86,11 +181,39 @@ public void acceptHttpConnectionTest() throws IOException {
 		conn.setRequestProperty("ServiceBusAuthorization", tokenString);
 
 		assertEquals("Response did not have the expected response code.", status, conn.getResponseCode());
-		assertTrue("Listener failed to accept connections from sender in http mode.", checkHttpConnectionTask.join());	
+		assertTrue("Listener failed to accept connections from sender in http mode.", handlerExecuted.get());
+		
+		listener.close();
+	}
+	
+	@Test
+	public void listenerReconnectionTest() {
+		HybridConnectionListener listener = new HybridConnectionListener(CONNECTION_URI, tokenProvider);
+		AtomicBoolean handlerExecuted = new AtomicBoolean(false);
+		
+		listener.setConnectingHandler((ex) -> {
+			assertTrue("Exception should be ConnectionLostException.", ex != null && ex instanceof ConnectionLostException);
```

Comment:
>assertTrue [](start = 3, length = 10)

I'm not sure asserting and throwing in an event handler is a good idea. Some would argue that's a violation of the implicit contract of an event handler.